db/db_impl.cc

```diff
@@ -645,6 +668,7 @@ Status DBImpl::SyncWAL() {
     need_log_dir_sync = !log_dir_synced_;
   }
 
+  TEST_SYNC_POINT("DBWALTest::SyncWALNotWaitWrite:1");
```

Comment:
Internet didn't say any thread safety guarantee them, so I assume it is safer for any operation to be locked, unless two threads are only doing reads.